File: code/Joyanta-Kumar/Structure/composition/Graph.java
package composition;

import java.util.ArrayList;
import primitive.Edge;
import primitive.Node;

public class Graph {
    ArrayList<Node> nodes = new ArrayList<Node>();
    ArrayList<Edge> edges = new ArrayList<Edge>();

    public void show() {
        System.out.println("\"Nodes\": [");
        for (Node node : nodes) {
            System.out.print("\t");
            node.show();
            if (node != this.nodes.getLast()) {
                System.out.print(",");
            }
            System.out.println();
        }
        System.out.println("]");

        System.out.println("\"Edges\": [");
        for (Edge edge : edges) {
            System.out.print("\t");
            edge.show();
            if (edge != this.edges.getLast()) {
                System.out.print(",");
            }
            System.out.println();
        }
        System.out.println("]");
    }

    public void addNode(Node node) {
        this.nodes.add(node);
    }

    public void addEdge(Edge edge) {
        this.edges.add(edge);
    }
}

File: code/Joyanta-Kumar/Visual/modules/graph.py
class Graph:
    def __init__(self, nodes = [], edges = []):
        self.edges = edges
        self.nodes = nodes
    
    def draw(self, window):
        for edge in self.edges:
            edge.draw(window)
        for node in self.nodes:
            node.draw(window)
    
    def addNode(self, node):
        if node not in self.nodes:
            self.nodes.append(node)
    
    def addEdge(self, edge):
        if edge not in self.edges:
            self.edges.append(edge)